src/triage/component/catwalk/predictors.py

```diff
@@ -151,75 +147,100 @@ def _write_to_db(
         self._existing_predictions(session, model_id, matrix_store)\
             .delete(synchronize_session=False)
         session.expire_all()
-        db_objects = []
         test_label_timespan = matrix_store.metadata['label_timespan']
         logging.warning(test_label_timespan)
+        session.commit()
+        session.close()
```

Comment:
So long as we're fixing this, I think we'll want try/finally –

    try:
        // do things with session
        // ...
        session.commit()
    except:
        session.rollback()
        raise
    finally:
        session.close()

Or let's define a context-management helper:

    from contextlib import contextmanager

    @contextmanager
    def session_manager(constructor):
        """Provide a transactional scope around a series of operations."""
        session = constructor()
        try:
            yield session
            session.commit()
        except:
            session.rollback()
            raise
        finally:
            session.close()

For this class we could say:

    @property
    def session(self):
        return session_manager(self.sessionmaker)

And here we could easily and safely say:

    with self.session as session:
        self._existing_predictions(session, …)
        … 

(See also sqlalchemy [session docs](http://docs.sqlalchemy.org/en/latest/orm/session_basics.html#when-do-i-make-a-sessionmaker).)